01-introduction.Rmd

```diff
@@ -45,3 +37,47 @@ What to expect in this book:
 - **Chapter \@ref(c10-ambarom-vignette)**: AmericasBarometer Vignette. A vignette on how to analyze data from the AmericasBarometer, a survey of attitudes, evaluations, experiences, and behavior in countries in the Western Hemisphere. This includes how to make choropleth maps with survey estimates.
 
 Most chapters include code to follow along with.  Each chapter with this code will start with a set-up section.  This will include all of the code needed to load packages and datasets to use in the chapter.  We then provide an overview of the topic along with examples on how to use the functions.  Most chapters end with exercises to work through.  Solutions to these exercises can be found online at [XX].
+
+The two main datasets we use throughout the book are the Residential Energy Consumption Survey [RECS -- @recs-2015-micro] and the American National Election Studies [ANES -- @debell].  To ensure that all readers can easily follow along with the examples, we have provided analytic datasets for readers on OSF^[https://osf.io/gzbkn/?view_only=8ca80573293b4e12b7f934a0f742b957]. To load any data used in the book that is not included in existing packages, we have created a helper function `read_osf()`. Here is the code to read in RECS and ANES datasets: 
+
+```{r}
+#| label: intro-setup
+#| error: FALSE
+#| warning: FALSE
+#| message: FALSE
+library(tidyverse)
+library(survey)
+library(srvyr)
+library(osfr)
+source("helper-fun/helper-functions.R")
+```
+
+```{r}
+#| label: intro-setup-readin
+#| error: FALSE
+#| warning: FALSE
+#| message: FALSE
+#| cache: TRUE
+recs_in <- read_osf("recs_2015.rds")
+anes_in <- read_osf("anes_2020.rds")
+```
+
+
+RECS is a household study of energy consumption in the U.S. which provides energy consumption and expenditures data. RECS funded by Energy Information Administration and collects information through energy suppliers through in-person, phone, and web interviews. It has been fielded 14 times between 1950 and 2020 and includes questions about appliances, electronics, heating, air conditioning (A/C), temperatures, water heating, lighting, energy bills, respondent demographics, and energy assistance.  Here is an overview of the data we just read in `recs_in`:
+
+```{r}
+#| label: intro-recs
+recs_in
+```
+
+From this output we can see that there are `r nrow(recs_in) %>% formatC(big.mark = ",")` rows and `r ncol(recs_in) %>% formatC(big.mark = ",")` variables.  We can see that there are variables containing an ID (`DOEID`), regional information (e.g., `Region`, `MSAStatus`), along with information about the house including the type of house (`HousingUnitType`) and when the house was built (`YearMade`). Additionally, there are a long list of weighting variables that will be used in the analysis (e.g., `NWEIGHT`, `BRRWT1`).  We will discuss how to use these weighting variables in Chapter \@ref(c03-specifying-sample-designs).
```

Comment:
Do we want to create codebooks for the data we create here? maybe as an appendix?  thinking for both RECS and ANES.